jsettlers.graphics/src/jsettlers/graphics/utils/UIList.java

```diff
@@ -49,19 +49,18 @@
 		UIListItem getItem(T item);
 	}
 
-	public UIList(List<? extends T> items, ListItemGenerator<T> generator,
-			float itemheight) {
-		this.items = items;
+	public UIList(List<? extends T> items, ListItemGenerator<T> generator, float itemheight) {
+		setItems(items);
 		this.generator = generator;
 		this.itemheight = itemheight;
-		if (items.size() > 0) {
-			activeItem = items.get(0);
-		}
 	}
 
 	public void setItems(List<? extends T> list) {
 		synchronized (itemsMutex) {
 			this.items = list;
+			if (items.size() > 0) {
```

Comment:
I think maybe you meant that you want the previous selection to be kept when going back to a menu. I've updated it now so the active item for each menu is not reset when clicking between menus.